### PR TITLE
Console warning

### LINKED ISSUE #15: Security: Add Self XSS Warning to Dev Console
Add a warning in the Javascript console similar to this example from www.facebook.com
image

### PR DESCRIPTION
**Pull request by tfroseman**
Added warning to console. Fixes #15
screen shot 2017-02-27 at 6 14 26 pm

### CODE REVIEW

**Comment by umurkontaci** [comment]
`console.log` is not initialized on all browsers until the devtools is actually opened. We should check for the existence of console.

Not all browsers support color input. We should determine if the console supports color and print according.

We should i18nize the message.

I guess for added security, we could run the console call in a setTimeout or try-catch so any error doesn't prevent the rest of the execution. 

What do you think @mtias ?

**Comment by dmsnell** [comment]
@tfroseman are you still interested in pushing out this PR?

**Comment by tfroseman** [comment]
Definitely. Wasn't certain if I needed input from @mitas. If not I can go about adding the desired checks for console and errors.

**Comment by dmsnell** [comment]
awesome - thanks @tfroseman! yeah I would suggest going ahead and addressing what @umurkontaci  raised.

@umurkontaci do we have numbers on that behavior for `console.log()` or colors not being available? I would hate to get too deep into it and find out that only our "non supported" browsers are the ones causing trouble.

likewise I'm curious about translation since this is just an eval'd JS string in the HTML header. if we wanted to translate it seems like we would need a lot more code (maybe it belongs in the app and not the HTML header…?)

**Comment by umurkontaci** [comment]
> do we have numbers on that behavior for console.log() or colors not being available?

I think IE < 10 doesn't have console before you open the dev tools, it could be the same in IE > 10 as well. I don't have any data on how browsers behave regarding colored console commands when they don't support it.

But this will execute very early and on every load, so we can't have any errors thrown. 

Although I think it might be a better to fit to run this after the app is loaded. That way we'll have access to translation module and error capturing.

File: server/pages/index.jade [hunk 1]
```diff
@@ -133,5 +133,23 @@ html(lang=lang, dir=isRTL ? 'rtl' : 'ltr', class=isFluidWidth ? 'is-fluid-width'
...
+					}else{
+						//No color supported
+                        console.log("Wait! This area is meant for developers!");
+					}
+                    console.log("If someone told you to type or paste something into this area it could compromise the security of your account.");
+
+                }
+			} catch (exception) {
+					// Do something with error
+            }
```
Comment (dmsnell) [inline server/pages/index.jade]: why the `try/catch` if we are also checking for the presence of `window.console`? seems like we would only need one or the other

Comment (tfroseman) [inline server/pages/index.jade]: Made sense at the time as a way to meet the the need for error capturing if needed this early in app loading. Not really sure what errors could/would get thrown from writing to console.

**Comment by dmsnell** [comment]
@tfroseman what's the status on this PR? would you like to continue to refine it or have you moved on? are you waiting on feedback from one of us?

**Comment by tfroseman** [comment]
@dmsnell Wasn't sure if the ability to support localization would fall under the scope of this issue. With that said I would like to have this added, I'll remove the redundant check.

**Comment by dmsnell** [comment]
I'm all for it. @yoavf @mtias or @ehg any thoughts on this? seems like waiting for localization would put a big hold on this PR since it's in the initial HTML file. can we get a 👍 or a 👎 vote for if you think this warning adds enough value to add, and to do so in English-only?

File: server/pages/index.jade [hunk 2]
```diff
@@ -142,5 +142,20 @@ html(lang=lang, dir=isRTL ? 'rtl' : 'ltr', class=isFluidWidth ? 'is-fluid-width'
...
+				if(true) {
+                    //Browser supports color output
+                    console.log("%cWait! This area is meant for developers!", "color: red;font-size: xx-large");
+				}else{
+					//No color supported
+                    console.log("Wait! This area is meant for developers!");
+				}
+                console.log("If someone told you to type or paste something into this area it could compromise the security of your account.");
+
+            }
```
Comment (dmsnell) [inline server/pages/index.jade]: @tfroseman do you think you could fix the formatting issues here? I think that if we get that updated we can ship. We will also prefer single quotes on the strings.

Comment (tfroseman) [inline server/pages/index.jade]: @dmsnell Yeah I'll clean that up. Also I was testing console on safari and realized that it and IE do not support 'rich' console output. Can I just do agent sniffing and only output plain text to log?

Comment (dmsnell) [inline server/pages/index.jade]: Seemed to work in Safari. What specific troubles did you find there? We'd probably prefer not to try and sniff browser agent.

If it's a big problem we can probably start by leaving it all black.

Comment (tfroseman) [inline server/pages/index.jade]: screen shot 2017-06-26 at 12 48 11 pm
The css styled output.
I am on Version 10.0.1 (11602.2.14.0.7) if that makes any difference.

Comment (dmsnell) [inline server/pages/index.jade]: I'm on 10.1.1 and that's working. I'm fine leaving it in for Safari. IE is the bigger question.

**Comment by tfroseman** [comment]
@dmsnell You are correct works on everything but IE as it does not yet support css in console.

**Comment by gziolo** [comment]
There is also similar solution proposed by @kloon with #18088.

**Comment by dmsnell** [comment]
Resolved with the merge of #18088

**Pull request closed without merging**